### DISCUSSION TITLE
smatweight/grouplist

### DISCUSSION THREAD

**Discussion by Rana-Phy** [FitSNAP/FitSNAP#84] [Q&A] [answered]
Dear Devolopers,

Greetings. I am searching for information on smatweight in this repo.
Would you kindly share how to use it with the latest version of fitsnap3?
(The **template.in** file in the doc folder, is it for this version? Do you suggest using the older version of fitsnap?)
.................................................................................................................................................
[REFERENCE]
units = metal
atom_style = atomic

pair_style= zero 10.0
pair_coeff= * *

[GROUPS]
group_sections = name training_size testing_size eweight fweight vweight
group_types = str float float float float float
smartweights = 1
Ti_eos     	= 1.0 0.0 **100 1 1e-4**

**Comment by mitwood** [comment, marked as answer]
Hey @Rana-Phy ,

Smartweights are a tool to make it easier on users to ensure that all of their training groups are equally valued when preforming linear regression. Many times training data comes in different numbers of configurations, as well as different atoms per configuration. These would normally cause the larger groups to dominate the fit. For example, the Ti_eos group may have just one training configuration with 2 atoms. Meanwhile in another group, called AIMD for example, there were 100 configurations with 100 atoms apiece.

The group section would look like:
[GROUPS]
group_sections = name training_size testing_size eweight fweight vweight
group_types = str float float float float float
smartweights = 1
Ti_eos = 1.0 0.0 100 1 1e-4
AIMD = 1.0 0.0 100 1 1e-4

The Ti_eos group contributes 1 energy, 3 * 2(atoms) * 1(configs) forces and 6 * 1(configs) stress training points to the fit.
The AIMD group contributes 100 energy, 3 * 100 * 100 forces and 6 * 100 stress training points to the fit.

If these two groups of training are to be considered equal in determining the SNAP coefficients, then a normalizing factor needs to be applied to the weights that are present (i.e. 100 1 1e-4) that is proportional to the number of 1/configurations, 1/number_of_forces and 1/configurations. This normalizing factor is applied on top of the user defined weights. 

Feel free to follow up, I know this topic isn't very clear at first attempt.

-Mitch

**Reply by Rana-Phy** [reply]
Dear Dr @mitwood, 

Thanks for reply. I have almost got your explanation.
To be clear I want to know more.
As above example,
Why for both eos and aimd group, it should be '100 1 1e-4' when both group containing  different number of structures and atoms?
(For Ti_eos shouldn’t we use '1 1 1e-4' since it is containg only one configuration? Does '100 1 1e-4' set have any specific physics meaning?)

According to the doc/Template.in, if e/f/v weight are defined these will be used as numerators of 1/(number of e/f/v rows).
Is that mean that setting smartwieght is normalizing the weights for different groups. I mean does smartweigt=1 is the normalization factor you described in the last para of your previous answer?

I think it will be very helpful for other users and me if you kindly explain one more time the
' 1 1 1' case vs '1*N 1 1e-4' case.

Best regard,
Rana

**Reply by mitwood** [reply]
"I mean does smartweights=1 is the normalization factor you described in the last para of your previous answer?"
> Yes, exactly. The denominator is the normalizing factor applied when smartweights are used, and the numerator is whatever is given as the eweight, fweight, vweight for that specific group. 

Therefore if you wanted all energies, forces and stresses from all training groups(regardless of how many training points each contributes) to be valued equally in a fit you would set:

[GROUPS]
group_sections = name training_size testing_size eweight fweight vweight
group_types = str float float float float float
smartweights = 1
Group1 = 1.0 0.0 1 1 1
Group2 = 1.0 0.0 1 1 1
Group3 = 1.0 0.0 1 1 1

Where the "1 1 1" is re-enforcing of 'equal-value' when smartweights =1. 

My advice is that you will find that the best fits for your interatomic potentials are where "1 1 1" for all groups is not desirable, and these can be treated as free variables during an optimization. "100 1 1E-4" from the first example you sent implies that energies are 100 times more important than forces, and stresses even less so. Changing the ratio of eweight:fweight while keeping weight low is a good place to start from my experience.

**Reply by Rana-Phy** [reply]
Dear Dr. @mitwood ,

Thanks for clarifying the things. I am trying with different eweight:weight. 
I will write about my experience.
Thanks.
